packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -58,13 +64,71 @@ export async function fetchSnapshot(
     );
 }
 
+export async function redeemSharingLink(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+) {
+    await getWithRetryForTokenRefresh(async (tokenFetchOptions) => {
```

Comment:
Are we reporting perf metrics (i.e. wrapping with PerformanceEvent)?

Reply:
We can report. So we had a metric which report total time including fallbacks if any and then a metric  which measures just the core trees/latest, For now I can add one, just to measure how much this takes.